dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/storage/EventDAO.scala

```diff
@@ -0,0 +1,71 @@
+package org.bitcoins.dlc.oracle.storage
+
+import org.bitcoins.crypto.{FieldElement, SchnorrNonce}
+import org.bitcoins.db.{AppConfig, CRUD, DbCommonsColumnMappers, SlickUtil}
+import org.bitcoins.dlc.oracle.SigningVersion
+import slick.lifted.{ForeignKeyQuery, ProvenShape}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class EventDAO()(implicit
```

Comment:
So in my opinion we should have a unique DAO for each `SigningVersion` as we might not know what the requirements are for signing? This reminds me of the situation we have with spending info table (what we need to sign a segwit v1 utxo is very different than a segwit v0 output) -- should we apply this here as well?

Reply:
We shouldn't need a DAO for each `SigningVersion`. Since the `EventDAO` doesn't contain the outcomes instead the `EventOutcomeDAO` does we can easily add on more fields if needed without effecting older outcomes. If we really needed to we could make a `EventOutcomeV2DAO` but I think how it stands currently is forwards compatible enough.